mindmeld/components/entity_resolver.py

```diff
@@ -1693,6 +1381,63 @@ def _load(self):
         self.fit()
 
 
+class SentenceBertCosSimEntityResolver(EmbedderCosSimEntityResolver):
+    """
+    Resolver class for bert models as described here:
+    https://github.com/UKPLab/sentence-transformers
+    """
+
+    def __init__(self, app_path, **kwargs):
+        """
+        This wrapper class allows creation of a BERT base embedder class
+        (currently based on sentence-transformers)
+
+        Specificall, this wrapper updates er_config in kwargs with
+            - any default settings if unavailable in input
+            - cache path
+
+        Args:
+            app_path (str): App's path to cache embeddings
+            er_config (dict): Configurations can be passed in through `model_settings` field
+                `model_settings`:
+                    embedder_type: the type of embedder picked from embedder_models.py class
+                        (eg. 'bert', 'glove', etc. )
+                    pretrained_name_or_abspath: the pretrained model for 'bert' embedder
+                    bert_output_type: if the output is a sentence mean pool or CLS output
+                    quantize_model: if the model needs to be quantized for faster inference time
+                        but at a possibly reduced accuracy
+                    concat_last_n_layers: if some of the last layers of a BERT model are to be
+                        concatenated for better accuracies
+                    normalize_token_embs: if the obtained sub-token level encodings are to be
+                        normalized
+        """
+
+        # default configs useful for reusing model's encodings through a cache path
+        defaults = {
+            "embedder_type": "bert",
+            "pretrained_name_or_abspath": "distilbert-base-nli-stsb-mean-tokens",
+            "bert_output_type": "mean",
+            "quantize_model": True,
+            "concat_last_n_layers": 4,
+            "normalize_token_embs": True,
+        }
+
+        # determine cache path and pass it in `model_settings`
+        er_config = copy.deepcopy(kwargs.get("config", {}))
+        model_settings = er_config.get("model_settings", {})
```

Comment:
my understanding is that it would be possible to specify other embedder model to use in the `embedder_type` param and not necessarily BERT for the resolver, while this resolver model class is BERT specific?

Reply:
Thats correct. Although we advertise BERT based embedder in docs, a generic embedder based class is created to aid QA non-ES class.